src/Apps/Order2/Routes/Checkout/CheckoutContext/Order2CheckoutContext.tsx

```diff
@@ -616,7 +617,33 @@ const CheckoutLoadingManager: React.FC<{
   useEffect(() => {
     const timeout = setTimeout(() => {
       setMinimumLoadingPassed(true)
-    }, MINIMUM_LOADING_MS)
+    }, MIN_LOADING_MS)
+    return () => clearTimeout(timeout)
+  }, [])
+
+  const isLoadingRef = useRef(isLoading)
+  useEffect(() => {
+    isLoadingRef.current = isLoading
+  }, [isLoading])
+
+  // biome-ignore lint/correctness/useExhaustiveDependencies: <explanation>
+  useEffect(() => {
+    const timeout = setTimeout(() => {
+      if (isLoadingRef.current) {
+        logger.error(
+          `Checkout loading state exceeded ${MAX_LOADING_MS}ms timeout: ${Object.entries(
```

Comment:
Is this more of a temporary logging for diagnosing the reported issue? Curious if there could be errors elsewhere that prevent this from working.

Separately it seems useful to make sure errors are properly captured in Sentry for future diagnosis, which I'm not sure the current status for Force. Non-blocking for this PR.

Reply:
Using `logger.error()` I believe [does include sentry logging if provided an error class](https://github.com/artsy/force/blob/d27fe594240303965e79e121fbc6d0f7d2c252e5/src/Utils/logger.ts#L21-L24). I went to double check this and caught the second part, so now it should be fixed.

Reply:
I went with this approach because we've had reports on this (mainly in eigen) but haven't been able to reliably reproduce it.